runner/app/routes/audio_to_text.py

```diff
@@ -14,39 +15,47 @@
 
 logger = logging.getLogger(__name__)
 
-RESPONSES = {
-    status.HTTP_400_BAD_REQUEST: {"model": HTTPError},
-    status.HTTP_401_UNAUTHORIZED: {"model": HTTPError},
-    status.HTTP_413_REQUEST_ENTITY_TOO_LARGE: {"model": HTTPError},
-    status.HTTP_500_INTERNAL_SERVER_ERROR: {"model": HTTPError},
-}
-
 
 def handle_pipeline_error(e: Exception) -> JSONResponse:
-    """Handles exceptions raised during audio processing.
+    """Handles exceptions raised during audio pipeline processing.
 
     Args:
         e: The exception raised during audio processing.
 
     Returns:
         A JSONResponse with the appropriate error message and status code.
     """
-    logger.error(f"Audio processing error: {str(e)}")  # Log the detailed error
+    logger.error(f"AudioToText pipeline error: {str(e)}")  # Log the detailed error
     if "Soundfile is either not in the correct format or is malformed" in str(
         e
     ) or isinstance(e, AudioConversionError):
         status_code = status.HTTP_415_UNSUPPORTED_MEDIA_TYPE
         error_message = "Unsupported audio format or malformed file."
+    elif "CUDA out of memory" in str(e) or isinstance(e, OutOfMemoryError):
+        status_code = status.HTTP_400_BAD_REQUEST
+        error_message = "Out of memory error."
+    elif isinstance(e, InferenceError):
+        status_code = status.HTTP_400_BAD_REQUEST
+        error_message = str(e)
```

Comment:
Check if errors are forwarded to the client.

Reply:
Just commited my stashed changes to this pull request. Should be fixed now. The only error that is not correctly handled yet is the 422 error. We should add validation at the go-livepeer side (See https://linear.app/livepeer-ai/issue/AI-628/correctly-handle-422-errors) and add a FastAPI validation error exception handler.